minlzlib/xzstream.c

```diff
@@ -378,31 +399,31 @@ XzDecodeBlockHeader (
     Container.HeaderSize = (blockHeader->Size + 1) * 4;
     if (Container.HeaderSize != sizeof(*blockHeader))
     {
-        return false;
+        return XzBlockHeaderFail;
     }
 
     //
     // Validate that no additional flags or filters are enabled
     //
     if (blockHeader->u.Flags != 0)
     {
-        return false;
+        return XzBlockHeaderFail;
     }
 
     //
     // Validate that the only filter is the LZMA2 filter
     //
     if (blockHeader->LzmaFlags.Id != 0x21)
```

Comment:
I had a personal todo to replace the magic with a `const uint8_t k_XzLzma2FilterIdentifier = 0x21`, if you can handle it, I'd appreciate it :)

Reply:
done.